cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpCredentialConnector.java

```diff
@@ -72,16 +66,148 @@ public CloudCredentialStatus create(AuthenticatedContext authenticatedContext) {
+        } catch (Exception err) {
+            LOGGER.debug("Could not parse TokenResponseException", err);
+            return Optional.empty();
         }
     }
 
-    @Override
-    public CloudCredentialStatus delete(AuthenticatedContext authenticatedContext) {
-        return new CloudCredentialStatus(authenticatedContext.getCloudCredential(), CredentialStatus.DELETED);
+    /**
+     * Attempts to obtain the "error_description" string parameter from the
+     * provided JsonObject as a String Optional. If the provided JsonObject
+     * is null, or there is no "error_description" element in the JsonObject,
+     * than an empty Optional going to return.
+     *
+     * @param tokenResponseExceptionAsJson the actual JsonObject which should
+     *                                     contains a String key called
+     *                                     "error_description".
+     * @return If
+     */
+    private Optional<String> getErrDescriptionFromTokenErrorJson(JsonObject tokenResponseExceptionAsJson) {
+        Optional<String> message = Optional.empty();
+        try {
+            message = Optional.of(tokenResponseExceptionAsJson.get("error_description").getAsString());
+        } catch (Exception elementNotFound) {
+            LOGGER.debug("Could not obtain the \"error_description\" element from the given Json", elementNotFound);
+        }
+        return message;
+    }
+
+    /**
+     * Checks both the provided GcpContexts's project id, service account id
+     * and compute, to be sure the given context is valid for further
+     * operations.
+     *
+     * @param gcpContext the GcpContext which should contain a valid project
+     *                   id, service account id and compute instance
+     */
+    private void checkGcpContextValidity(GcpContext gcpContext) {
+        checkProjectId(gcpContext);
+        checkServiceAccountId(gcpContext);
+        checkCompute(gcpContext);
     }
+
+    /**
+     * Checks the GcpContext's project id. If there is no id, or it's size is
+     * 0, than exception would be thrown.
+     *
+     * @param context the inspected GcpContext instance which should contains
+     *                a valid project id.
+     */
+    private void checkProjectId(GcpContext context) {
+        if (context == null || !StringUtils.hasLength(context.getProjectId())) {
+            throw new CloudConnectorException("Project id is missing.");
+        }
+    }
+
+    /**
+     * Checks the GcpContext's service account id. If there is no id, or it's
+     * size is 0, than exception would be thrown.
+     *
+     * @param context the inspected GcpContext instance which should contains
+     *                a valid service account id.
+     */
+    private void checkServiceAccountId(GcpContext context) {
+        if (context == null || !StringUtils.hasLength(context.getServiceAccountId())) {
+            throw new CloudConnectorException("Service account id is missing.");
```

Comment:
Not the best practice to throw unchecked runtime exception from here. Checked exception or some return value would be better.